paasta_tools/api/client.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Client interface for the Paasta rest api.
+"""
+import json
+import logging
+import os
+import sys
+
+from bravado.client import SwaggerClient
+
+from paasta_tools.utils import load_system_paasta_config
+
+log = logging.getLogger(__name__)
+
+
+def get_paasta_api_client(cluster=None, system_paasta_config=None):
+    if not system_paasta_config:
+        system_paasta_config = load_system_paasta_config()
+
+    if not cluster:
+        cluster = system_paasta_config.get_cluster()
+
+    api_endpoints = system_paasta_config.get_api_endpoints()
+    if cluster not in api_endpoints:
+        log.error('Cluster %s not in paasta-api endpoints config', cluster)
+        return None
+
+    url = str(api_endpoints[cluster])
+    if not url.startswith("http://"):
+        log.error('Unsupported paasta-api url %s', url)
+        return None
+
+    api_server = url[7:]
+
+    # Get swagger spec from file system instead of the api server
```

Comment:
I can't say I'm expert enough to know the right option here. Maybe you've thought it through or someone has an opinion? I guess this couples the client to a specific version of the API rather than whatever is running at the endpoint it hits. However, maybe that's actually safer? Happy either way unless anyone has any more detailed thoughts.